### ISSUE TITLE
Incomplete diagnostic message

### ISSUE THREAD

**Issue by hunterhogan** [microsoft/pyright#11125]
A diagnostic error without a message is a bug. A diagnostic error with a message that has incorrect information is a bug. Due to the nature of a type checker, poorly written messages are bugs.

The following, for example, is a bug.

```python
Expression with type "tuple[tuple[tuple[int, int], tuple[int, int]], int]" cannot be assigned to target tuple
  Type "tuple[tuple[tuple[int, int], tuple[int, int]], int]" is incompatible with target tuple
    Tuple size mismatch; expected 3 but received 2
```

The semicolon, for example, is indisputably wrong; it is not joining two independent phrases: refining details follow. The same processes by which logical bugs are detected and removed from the code can be used to detect and remove bugs from the messages.

The message does not help identify the problem in my code:

```python
cc: Iterator[tuple[tuple[tuple[int, int], tuple[int, int]], int]] = CartesianProduct(combinations(sorted(pinnedLeavesFiltered.items()), 2), range(state.dimensionsTotal))
pp: Iterator[tuple[tuple[tuple[int, int], tuple[int, int]], int]] = filter(matchingParityIndexLeaf(state.mapShape), cc)
nn = [(pile, nextCreaseIndexLeaf(state.mapShape, indexLeaf, dimension), pileComparand, comparand, dimension) for ((pile, indexLeaf), (pileComparand, comparand), dimension) in pp]
```

I want to work with Microsoft to fix these bugs. And I want healthcare for my untreated illnesses so I can stop being homeless. I will prioritize activities that might improve my life, you know?

**Comment by erictraut** [comment]
I don't consider this a bug, so I'm closing it.

Sorry to hear that you're struggling with mental illness.

Incidentally, this is a Microsoft project, but I am its primary maintainer. I worked for Microsoft at the time that I initially wrote pyright, but no longer do. I contribute to it in my spare time. Pylance, which is built upon pyright, has a dedicated team of Microsoft engineers.

**Comment by hunterhogan** [comment]
Thank you for the info. 

Most especially, thank you for your time that you generously give to an inconceivable number of people.
